tests/extra_source_and_tests/user_2d_T_shaped_pipe_VIPO_shell/T_shaped_pipe_VIPO_shell.cpp

```diff
@@ -0,0 +1,554 @@
+            addSurfaceProperties(normal_direction, wall_thickness_);
+        }
+    }
+};
+
+//----------------------------------------------------------------------
+//	Inflow velocity
+//----------------------------------------------------------------------
+struct InflowVelocity
+{
+    Real u_ref_, t_ref_;
+    AlignedBoxShape &aligned_box_;
+
+    template <class BoundaryConditionType>
+    InflowVelocity(BoundaryConditionType &boundary_condition)
+        : u_ref_(U_f), t_ref_(2.0),
+            aligned_box_(boundary_condition.getAlignedBox()){}
+
+    Vecd operator()(Vecd &position, Vecd &velocity)
+    {
+        Vecd target_velocity = Vecd::Zero();
+        
+        Real run_time = GlobalStaticVariables::physical_time_;
+        Real u_ave = run_time < t_ref_ ? 0.5 * u_ref_ * (1.0 - cos(Pi * run_time / t_ref_)) : u_ref_;
+        target_velocity[0] = u_ave;
+        target_velocity[1] = 0.0;
+
+        return target_velocity;
+    }
+};
+
+class TimeDependentAcceleration : public Gravity
+{
+    Real t_ref_, u_ref_, du_ave_dt_;
+
+  public:
+    explicit TimeDependentAcceleration(Vecd gravity_vector)
+        : Gravity(gravity_vector), t_ref_(2.0), u_ref_(0.0), du_ave_dt_(0) {}
+
+    virtual Vecd InducedAcceleration(const Vecd &position) override
+    {
+        Real run_time_ = GlobalStaticVariables::physical_time_;
+        du_ave_dt_ = 0.5 * u_ref_ * (Pi / t_ref_) * sin(Pi * run_time_ / t_ref_);
+
+        return run_time_ < t_ref_ ? Vecd(du_ave_dt_, 0.0) : global_acceleration_;
+    }
+};
+
+/**
+ * @brief 	Pressure boundary definition.
+ */
+struct LeftInflowPressure
+{
+    template <class BoundaryConditionType>
+    LeftInflowPressure(BoundaryConditionType &boundary_condition) {}
+
+    Real operator()(Real &p_)
+    {
+        return p_;
+    }
+};
+
+struct UpOutflowPressure
+{
+    template <class BoundaryConditionType>
+    UpOutflowPressure(BoundaryConditionType &boundary_condition) {}
+
+    Real operator()(Real &p_)
+    {
+        /*constant pressure*/
+        Real pressure = Outlet_pressure;
+        return pressure;
+    }
+};
+
+struct DownOutflowPressure
+{
+    template <class BoundaryConditionType>
+    DownOutflowPressure(BoundaryConditionType &boundary_condition) {}
+
+    Real operator()(Real &p_)
+    {
+        /*constant pressure*/
+        Real pressure = Outlet_pressure;
+        return pressure;
+    }
+};
+
+class BoundaryGeometry : public BodyPartByParticle
+{
+  public:
+    BoundaryGeometry(SPHBody &body, const std::string &body_part_name)
+        : BodyPartByParticle(body, body_part_name)
+    {
+        TaggingParticleMethod tagging_particle_method = std::bind(&BoundaryGeometry::tagManually, this, _1);
+        tagParticles(tagging_particle_method);
+    };
+    virtual ~BoundaryGeometry(){};
+
+  private:
+    void tagManually(size_t index_i)
+    {
+        if (base_particles_.ParticlePositions()[index_i][0] < -DL_sponge + buffer_width
```

Comment:
`(base_particles_.ParticlePositions()[index_i][0] < 0`

It needs to be fixed at the inflow velocity zone

Reply:
Sorry my bad, the inflow zone is the same as the emitter now, just keep it as it is